Core.Collectors/Web/HttpExceptionSignature.cs

```diff
@@ -1,29 +1,49 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Microsoft.CloudMine.Core.Collectors.Collector;
 using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Microsoft.CloudMine.Core.Collectors.Web
-{
-    public class HttpExceptionSignature
+{ 
+    public class HttpExceptionSignature : IAllowListStatus
     {
-        public static readonly HttpExceptionSignature RequestTimeoutException = new HttpExceptionSignature(exception => 
+        public static HttpExceptionSignature RequestTimeoutException(Func<List<CollectionNode>> continuation = null)
         {
-            Type exceptionType = exception.GetType();
-            return exceptionType == typeof(TaskCanceledException) && exception.Message.Equals("The operation was canceled.");
-        });
+            Func<Exception, bool> matcher = exception =>
+            {
+                Type exceptionType = exception.GetType();
+                return exceptionType == typeof(TaskCanceledException) && exception.Message.Equals("The operation was canceled.");
+            };
+
+            return new HttpExceptionSignature(matcher, continuation);
+        }
 
         private readonly Func<Exception, bool> matcher;
+        private readonly Func<List<CollectionNode>> continuation;
 
-        public HttpExceptionSignature(Func<Exception, bool> matcher)
+        public HttpExceptionSignature(Func<Exception, bool> matcher, Func<List<CollectionNode>> continuation = null)
         {
             this.matcher = matcher;
+
+            if (continuation == null)
+            {
+                continuation = () => new List<CollectionNode>();
+            }
+
+            this.continuation = continuation;
         }
 
         public bool Matches(Exception exception)
         {
             return matcher.Invoke(exception);
         }
+
+        public List<CollectionNode> Continuation()
+        {
+            return this.continuation.Invoke();
```

Comment:
>return this.continuation.Invoke(); [](http://example.com/codeflow?start=12&length=34)

Nit: in general, I prefer seeing:

```
this.continuation();
```
(rather than explicit Invoke)

Reply:
trivia:
I will add on this, it's not only the matter of preference it actually "may" make a perf difference.
`.Invoke()` has the benefit of not caring about anything in terms of typing - to do this - it needs to do much more during execution instead of compile time. Invoke operates on `Object`s and `Binder`s and can execute "anything" out of thin air, while a regular call has the benefit of being generated properly by the compiler with all the necessary code but it requires all the details about the function to be known at compilation time.